app/ts/background/popupMessageHandlers.ts

```diff
@@ -177,29 +177,39 @@ export async function refreshPopupConfirmTransactionMetadata(ethereumClientServi
 	const promises = await getPendingTransactions()
 	if (promises.length === 0) return
 	const first = promises[0]
-	if (first.simulationResults.statusCode !== 'success') return
+	if (first.simulationResults === undefined || first.simulationResults.statusCode !== 'success') return
 	return await sendPopupMessageToOpenWindows({
 		method: 'popup_update_confirm_transaction_dialog',
 		data: [{
-			statusCode: 'success',
-			data: {
-				...first.simulationResults.data,
-				simulatedAndVisualizedTransactions: formSimulatedAndVisualizedTransaction(first.simulationResults.data.simulationState, first.simulationResults.data.visualizerResults, addressBookEntries),
-				addressBookEntries,
+			...first,
```

Comment:
I may be mistaken but seems this may be a `Promise.all`?

Reply:
hmm. I don't understand what do you mean?